### DISCUSSION TITLE
Add operation of MissionStateAction is hardcoded to plus mission state by 1

### DISCUSSION THREAD

**Discussion by A2ur3** [FakeFishGames/Barotrauma#14262] [Bug reports]
### Disclaimers

- [X] I have searched the issue tracker to check if the issue has already been reported.
- [X] My issue happened while using mods.

### What happened?

State attribute of MissionStateAction is useless when operation is set as Add although its description says "The state to set the mission to, or how much to add to the state of the mission."
And in vanilla event "clownhaven_progress" it uses value="1"

### Reproduction steps

1. Create a mission that uses progressbar
2. Trigger an event that adds missionstate by -1 every sec
3. Notice progress bar is increasing instead of decreasing

### Bug prevalence

Happens every time I play

### Single player or multiplayer?

Happens in both single player and multiplayer

### Version

v1.5.9.1 (Summer Update Hotfix 2)

### Which operating system did you encounter this bug on?

Windows

**Comment by ubertpendragon** [comment]
Looks like someone else tried to make a countdown using the progress bar

The progress bar (used by clown haven) has an extremely narrow implementation and can't be used in any other way in practice because it's using mission state, which is hardcoded (and not specified anywhere, Dak had to do some digging) and will eventually hit a number that either completes or fails a mission

![image](https://github.com/FakeFishGames/Barotrauma/assets/73229309/ae53c3d3-976c-456e-bf07-893e8807a70e)

**Comment by Regalis11** [comment]
Addressed in https://github.com/FakeFishGames/Barotrauma-development/commit/ac192415b1170bc799e8eb548d76586546283d99. Fix coming up in the next unstable build!
